lib/ssl/src/ssl_connection.hrl

```diff
@@ -92,8 +92,9 @@
 
 -record(connection_env, { 
                           user_application      :: {Monitor::reference(), User::pid()},
-                          downgrade,
-                          terminated = false                          ::boolean() | closed,  
+                          downgrade             :: {NewController::pid(), Caller::gen_statem:from()} | 'undefined',
```

Comment:
If the element in the tuple are called NewController and Caller in the type spec why are the variable names not named the same? Now when matching the variables are named Pid and From but why not NewController and Caller instead?
Just a thought.

Reply:
Well, to make it consistent I changed it again to  NewController and From as From is very commonly used to refer to the caller when using the gen_* behaviours.  And changed the code accordingly!